pennylane/circuit_graph.py

```diff
@@ -647,7 +669,17 @@ def has_path(self, a, b):
         Returns:
             bool: returns ``True`` if a path exists
         """
-        return nx.has_path(self._graph, a, b)
+        if a == b:
+            return True
+
+        return (
+            len(
+                rx.dijkstra_shortest_paths(
+                    self._graph, self._graph.nodes().index(a), self._graph.nodes().index(b)
+                )
+            )
+            != 0
+        )
```

Comment:
just double checking here, but `RX` does not have a `has_path` equivalent? The reason I ask is that we previously added this because just checking that the path exists in NX ended up being much faster than _getting_ the shortest path.

But maybe this speed difference doesn't affect RX?

Reply:
Yep! This is one of those missing functions in RX comparing to NX. RX `dijkstra_shortest_path` is still faster that NX `has_path` (see the benchmarks) Moreover, I will keep updating the use of RX in PL with the next releases of RX in the future.